src/main/java/org/primefaces/component/cascadeselect/CascadeSelectRenderer.java

```diff
@@ -116,9 +118,18 @@ protected void encodeInput(FacesContext context, CascadeSelect cascadeSelect, St
         writer.endElement("div");
     }
 
-    protected void encodeLabel(FacesContext context, CascadeSelect cascadeSelect, String valueToRender) throws IOException {
+    protected void encodeLabel(FacesContext context, CascadeSelect cascadeSelect, List<SelectItem> itemList, String valueToRender) throws IOException {
         ResponseWriter writer = context.getResponseWriter();
-        String placeholder = LangUtils.isNotBlank(valueToRender) ? valueToRender : cascadeSelect.getPlaceholder();
+
+        String itemLabel = valueToRender;
+        if (LangUtils.isNotBlank(valueToRender)) {
```

Comment:
Actually we should not check for empty or null here, one might even have selectItem with value="" or value="#{null}"

Reply:
Fixed!